core/src/core-api/theming.js

```diff
@@ -0,0 +1,106 @@
+import { LuigiConfig } from '.';
+import { GenericHelpers } from '../utilities/helpers';
+/**
+ * Functions to use Luigi Core Theming features.
+ * @namespace Theming
+ */
+class LuigiTheming {
+  constructor() {
+    this.currentTheme;
+  }
+
+  /**
+   * Retrieves the available themes
+   * @memberof Theming
+   * @returns {promise} resolves an array of theming objects
+   * @since NEXTRELEASE
+   * @example
+   * Luigi
+   *  .theming()
+   *  .getAvailableThemes()
+   *  .then((themes) => {
+   *     // Logic to generate theme selector
+   *  });
+   */
+  async getAvailableThemes() {
+    return await LuigiConfig.getConfigValueAsync('settings.theming.themes');
+  }
+
+  /**
+   * Sets the current theming object
+   * @memberof Theming
+   * @param {Object} a theming object, has the same structure as retrieved from **getAvailableThemes**
+   * @since NEXTRELEASE
```

Comment:
Since NEXTRELEASE doesn't work for now could you please insert the version of Luigi manually everywhere?

Reply:
No, just because it does not work automatically, we still just manually search-and-replace all NEXTRELEASE with the release. I have no clue at with version it will be released, so it would be wrong to add something there.